### PR TITLE
fix custom vllm eval args

### PR DESCRIPTION
**Pull request by Yunnglin**
# PR type
- [x] Bug Fix
- [ ] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

# PR information

Fix eval dataset with vllm inference backend args.
Add following in `ms-swift/swift/llm/utils/vllm_utils.py`
```python
if self.repetition_penalty is None:
    self.repetition_penalty = 1.0
if self.top_p is None:
    self.top_p = 1.0
if self.top_k is None:
    self.top_k = -1
```

## Experiment results

Paste your experiment result here(if needed).

### CODE REVIEW

**Comment by Jintao-Huang** [comment]
vllm_utils.py可以还原一下不, 怎么改了这么多哇

File: swift/llm/utils/vllm_utils.py [hunk 1]
```diff
@@ -260,12 +270,24 @@ def __post_init__(self):
                 self.top_k = -1
             if self.stop is None:
                 self.stop = []
+            if self.repetition_penalty is None:
```
Comment (Jintao-Huang) [inline swift/llm/utils/vllm_utils.py]: 可以在eval.py中对None进行过滤不

Comment (Yunnglin) [reply swift/llm/utils/vllm_utils.py]: done

File: swift/llm/utils/vllm_utils.py [hunk 2]
```diff
@@ -620,5 +620,6 @@ def prepare_vllm_engine_template(args: InferArguments, use_async: bool = False)
         args.truncation_strategy,
         model=llm_engine,
         tools_prompt=args.tools_prompt)
-    logger.info(f'system: {template.default_system}')
+    args.system = template.default_system
```
Comment (Jintao-Huang) [inline swift/llm/utils/vllm_utils.py]: 这里是为什么呢

File: swift/llm/eval.py [hunk 3]
```diff
@@ -93,20 +107,34 @@ def predict(self, prompts: List[str], **kwargs) -> List[Dict[str, Any]]:
...
         elif self.args.infer_backend == 'vllm':
             from .utils import inference_vllm, VllmGenerationConfig
+
             if do_sample is False:
                 infer_cfg['temperature'] = 0
+            max_new_tokens = infer_cfg.pop('max_new_tokens', None)
+            if max_new_tokens is not None:
+                infer_cfg['max_tokens'] = max_new_tokens
+            defaults = {'repetition_penalty': 1.0, 'top_p': 1.0, 'top_k': -1}
+            # 使用默认值覆盖 None 值
```
Comment (Jintao-Huang) [inline swift/llm/eval.py]: please use english

File: swift/llm/utils/vllm_utils.py (hunk 2 above)
Comment (Yunnglin) [reply swift/llm/utils/vllm_utils.py]: It was likely modified by mistake and has been restored.

File: swift/llm/eval.py (hunk 3 above)
Comment (Yunnglin) [reply swift/llm/eval.py]: fixed

**Comment by Jintao-Huang** [comment]
thank you for your PR!

**Pull request merged**